bitcoin/src/p2p/mod.rs

```diff
@@ -243,31 +243,38 @@ impl FromStr for Magic {
     }
 }
 
-impl From<Network> for Magic {
-    fn from(network: Network) -> Magic {
-        match network {
-            // Note: new network entries must explicitly be matched in `try_from` below.
-            Network::Bitcoin => Magic::BITCOIN,
-            Network::Testnet => Magic::TESTNET,
-            Network::Signet => Magic::SIGNET,
-            Network::Regtest => Magic::REGTEST,
+macro_rules! generate_network_magic_conversion {
+    ($(Network::$network:ident => Magic::$magic:ident,)*) => {
+        impl From<Network> for Magic {
+            fn from(network: Network) -> Magic {
+                match network {
+                    $(
+                        Network::$network => Magic::$magic,
+                    )*
+                }
+            }
         }
```

Comment:
OMG, the diff renders horribly...

Reply:
git when it sees one matching curly brace: